types/chrome/index.d.ts

```diff
@@ -9565,7 +9565,7 @@ declare namespace chrome.tabs {
      * Sends a single message to the content script(s) in the specified tab, with an optional callback to run when a response is sent back. The runtime.onMessage event is fired in each content script running in the specified tab for the current extension.
      * @since Chrome 20.
      */
-    export function sendMessage<M = any, R = any>(tabId: number, message: M, responseCallback?: (response: R) => void): void;
+    export function sendMessage<M = any, R = any>(tabId: number, message: M, responseCallback: (response: R) => void): void;
```

Comment:
What the status on maintaining compatibility with MV2? This change breaks the definition for MV2 `sendResponse()` where it is necessary to pass `undefined` for the response callback to indicate that no async response is expected. Otherwise the runtime errors are reported.

Reply:
Chrome is in the process of removing all MV2 support, and is no longer supporting adding public mv2 extensions. See [here](https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/) for more details. My point is that there's no point in maintaining backwards compatibility.

Reply:
That's an unfortunate and premature stance. From your own link, developers have until June 2023 before the stable channel experiments with full MV2 deprecation. Developers need to maintain their MV2 extensions while working on MV3 transitions (which is not an easy lift in many cases). Count me in this group, as I'm waiting for [offscreen docs](https://bugs.chromium.org/p/chromium/issues/detail?id=1339382). It would sure be helpful to have MV2 types available until then. Consider also that line 9568 really is MV2-only. There is no MV3 signature like it, so that never should have been touched.

Reply:
If you run `chrome.runtime.onMessage.addListener((m, s, sr)=>sr('answer'))` in your current tab and `chrome.tabs.sendMessage((await chrome.tabs.query({active: true}))[0].id, 'message', console.log)` in the worker of a MV3 extension `'answer'` gets correctly logged to the console.
When using `undefined` in the arguments of `chrome.tabs.sendMessage()` instead of a callback function in a MV3 extension `chrome.tabs.sendMessage()` return a Promise.
However: If `undefined` gets used instead of a callback function in a MV2 extension `undefined` gets returned.

The aim of my PR was to update the types for chrome.tabs.sendMessage() to MV3 and callbacks are still a thing in MV3 so line **9568** and **9579** are not MV2-only.

I admit that the title of the PR is bad and that it should have been something like "Update chrome.tabs.sendMessage() types to MV3".

I also agree that MV2 is still relevant and important.
I forgot about backwards compatibility but I'd like to suggest to revert my changes to line **9568** and **9579** to maintain compatibility with MV2 even though that means that the types are incorrect for MV3.

Reply:
Some kind of feedback would be nice because I'm not going to "break" the MV3 types if no one wants me to do so.

Reply:
Because I did not receive any feedback for over a month now I consider this conversation resolved.